### PR TITLE
Add dev branch

### PR DESCRIPTION
**Pull request by suzhoum**
Issue #, if available:

Description of changes:

Adds an option for developers to test their working branch.
```
agbench run <config_file> --dev-branch <git_uri#branch_name>
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/autogluon/bench/entrypoint.sh [hunk 1]
```diff
@@ -1,3 +1,10 @@
...
 
-agbench run $config_file
+
+if [ -n "$AG_BENCH_DEV_URL" ]; then
+    echo "Using Development Branch: $AG_BENCH_DEV_URL" >&2
+    agbench run $config_file --dev-branch $AG_BENCH_DEV_URL
+else
+    echo "Using Released autogluon.bench: " >&2
+    agbench run $config_file
+fi
```
Comment (gidler) [inline src/autogluon/bench/entrypoint.sh:10]: Not a big deal, but we should add a new-line at the end of this file when possible.

**Comment by gidler** [comment]
Is there a reason we need to provide the ability to run a remote dev branch, rather than just letting them execute the version of the code they have locally? We could support this though copying the working directory into the image and then doing a `pip install -e .` or by building a wheel and copying it into the image for build. 

In the current solution, we require GitHub in the development loop when I'm not sure its necessary if all we need is to run working changes in the Dockerfile. Additionally, it might open up opportunity for weirdness if other people push code to the dev brach.

**Comment by suzhoum** [comment]
> Is there a reason we need to provide the ability to run a remote dev branch, rather than just letting them execute the version of the code they have locally? We could support this though copying the working directory into the image and then doing a `pip install -e .` or by building a wheel and copying it into the image for build.
> 
> In the current solution, we require GitHub in the development loop when I'm not sure its necessary if all we need is to run working changes in the Dockerfile. Additionally, it might open up opportunity for weirdness if other people push code to the dev brach.

The reason to support remove dev branch is because we are not able to do `pip install -e .`. The reason is that we are trying to install the package "within" the package itself.  `pip install -e .` is not possible without being under the root level of the package. That's why currently we can only do `pip install autogluon.bench==RELEASED_VERSION`.

**Comment by gidler** [comment]
> > Is there a reason we need to provide the ability to run a remote dev branch, rather than just letting them execute the version of the code they have locally? We could support this though copying the working directory into the image and then doing a `pip install -e .` or by building a wheel and copying it into the image for build.
> > In the current solution, we require GitHub in the development loop when I'm not sure its necessary if all we need is to run working changes in the Dockerfile. Additionally, it might open up opportunity for weirdness if other people push code to the dev brach.
> 
> The reason to support remove dev branch is because we are not able to do `pip install -e .`. The reason is that we are trying to install the package "within" the package itself. `pip install -e .` is not possible without being under the root level of the package. That's why currently we can only do `pip install autogluon.bench==RELEASED_VERSION`.

Is there a reason why we can't manually build a wheel file and install it in the container?

**Comment by suzhoum** [comment]
> > > Is there a reason we need to provide the ability to run a remote dev branch, rather than just letting them execute the version of the code they have locally? We could support this though copying the working directory into the image and then doing a `pip install -e .` or by building a wheel and copying it into the image for build.
> > > In the current solution, we require GitHub in the development loop when I'm not sure its necessary if all we need is to run working changes in the Dockerfile. Additionally, it might open up opportunity for weirdness if other people push code to the dev brach.
> > 
> > 
> > The reason to support remove dev branch is because we are not able to do `pip install -e .`. The reason is that we are trying to install the package "within" the package itself. `pip install -e .` is not possible without being under the root level of the package. That's why currently we can only do `pip install autogluon.bench==RELEASED_VERSION`.
> 
> Is there a reason why we can't manually build a wheel file and install it in the container?

I believe building a wheel require a `setup.py`? Then we need to have another setup.py inside the package if I'm not wrong. It might not be best practice to keep two `setup.py` for the project.

**Review by gidler — APPROVED** [review]

**Pull request merged**